docs/design/adrs/2024-05-02-enable-project-versioning.md

```diff
@@ -0,0 +1,63 @@
+
+* **Built-in features**: You can compare releases from the repository, see [Comparing releases](https://docs.github.com/en/repositories/releasing-projects-on-github/comparing-releases). You can also see the latest release from the repository homepage.
+
+* **Automation**: There is support to automatically generate [release notes](https://docs.github.com/en/repositories/releasing-projects-on-github/automatically-generated-release-notes) based on changes since the previous release, it also includes who has contributed.
+
+* **Simple**: There are extensive [docs](https://docs.github.com/en/repositories/releasing-projects-on-github/about-releases).
+
+* **Automated releases**: Using GitHub actions, it is also possible to automate the entire release, by generating a new version tag and creating a release with generated release notes.
+
+## Proposal
+
+- **Sprint Champion**: The Sprint Champion is responsible for reviewing any automated and manual releases which may occur during the sprint.
+- **Release cadence**:
+    - **Automated** releases will happen on each green PR merge. We will use GitHub actions to automate the release. This follows our current release cadence with the addition of adding a version. Note: Releasing smaller changesets is preferred in general since it is less riskier and easier to troubleshoot as compared to collecting changes for a week and releasing everything together.
+    - **Manual** releases will happen ad-hoc where required, such as for important bug fixes. These can be created manually and will be documented in the Release Guide.
+- **Release guide**: Create a Release Guide which includes what is required for automated and manual releases, define the versioning format (semantic versioning) and guidelines around what is considered a breaking change.
+- **How to automate**:
+    - When introducing automated versioning, there are a few things we need to consider:
+        - **major/minor/patch** versions: Each release is tagged with a semantic version which is based on the change included (e.g. breaking change, feature, bug fix).
+        - **Changelog**: A changelog must be included which details the changes that are included in the release.
+    - The [**release-please**](https://github.com/google-github-actions/release-please-action) GitHub Action is created by Google, and automates releases. It looks at commit messages to automatically generate the version number and to generate the changelog.
+        - It uses [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/), meaning we'd need to follow a convention of labelling PRs with `fix`/`feat`/`BREAKING CHANGE`, etc, in order to be picked up by the release generator. It uses these labels to create the correct version.
+        - As a consequence, current dependabot PRs would not trigger a release PR. These would be in main until a automatic or manual release was triggered.
```

Comment:
I really like this. The only small action we might need to take here is to enforce these prefixes on PR titles (fix/feat etc), otherwise we might inadvertently block releasing the latest changes. But a simple internet search shows multiple github actions that enforce rules on PR titles - so we can use that to enforce the prefixes.

Reply:
Great point, this made me realise we will also need to update the default dependabot PR titles, I have added a sub-task for that also. It also looks straightforward to do. Thanks for your input!